modules/platform/nuxeo-drive-server/nuxeo-drive-mongodb/pom.xml

```diff
@@ -0,0 +1,183 @@
+      <artifactId>nuxeo-core-management-jtajca</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.core</groupId>
+      <artifactId>nuxeo-core-management-jtajca</artifactId>
+      <scope>test</scope>
+      <type>test-jar</type>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-test</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.automation</groupId>
+      <artifactId>nuxeo-automation-test</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-userworkspace-types</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-userworkspace-core</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-collections-core</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-collections-core</artifactId>
+      <type>test-jar</type>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-relations-api</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-uidgen-core</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm</groupId>
+      <artifactId>nuxeo-drive-core</artifactId>
+      <type>test-jar</type>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.core</groupId>
+      <artifactId>nuxeo-core-cache</artifactId>
+      <type>test-jar</type>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.core</groupId>
+      <artifactId>nuxeo-core-redis</artifactId>
+      <type>test-jar</type>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-admin-center-core</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-webapp-types</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-search-core</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.nuxeo.ecm.platform</groupId>
+      <artifactId>nuxeo-platform-tag-core</artifactId>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+</project>
```

Comment:
There seems to be a lot of unused dependencies here.
Probably a copy-paste from another drive POM.
In this case, it's probably more relevant to start with a clean POM with no dependencies and add them as they're needed.

Don't hesitate to run `mvn -nsu dependency:analyze`, which currently shows:
```[INFO] --- maven-dependency-plugin:3.1.0:analyze (default-cli) @ nuxeo-drive-mongodb ---
[WARNING] Used undeclared dependencies found:
[WARNING]    org.nuxeo.ecm.core:nuxeo-core-query:jar:11.2-SNAPSHOT:compile
[WARNING]    org.mongodb:mongo-java-driver:jar:3.12.1:compile
[WARNING] Unused declared dependencies found:
[WARNING]    org.apache.commons:commons-lang3:jar:3.9:compile
[WARNING]    org.nuxeo.runtime:nuxeo-runtime-jtajca:jar:11.2-SNAPSHOT:compile
[WARNING]    org.nuxeo.ecm.core:nuxeo-core-event:jar:11.2-SNAPSHOT:compile
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-audit-core:jar:11.2-SNAPSHOT:compile
[WARNING]    org.nuxeo.ecm.automation:nuxeo-automation-core:jar:11.2-SNAPSHOT:compile
[WARNING]    org.nuxeo.ecm:nuxeo-drive-operations:jar:11.2-SNAPSHOT:compile
[WARNING]    javax.inject:javax.inject:jar:1:test
[WARNING]    org.nuxeo.ecm.core:nuxeo-core-management-jtajca:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.core:nuxeo-core-management-jtajca:test-jar:tests:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-test:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-userworkspace-types:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-userworkspace-core:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-collections-core:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-collections-core:test-jar:tests:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-relations-api:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-uidgen-core:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.core:nuxeo-core-cache:test-jar:tests:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.core:nuxeo-core-redis:test-jar:tests:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-admin-center-core:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-webapp-types:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-search-core:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-tag-core:jar:11.2-SNAPSHOT:test
```
So, many dependencies to remove and some to add.
Note, some dependencies required by the unit tests might not be detected since they're implicitly used by XML contributions, so you'll have to add them back if needed.

Reply:
Yes I copied nuxeo-drive-elasticsearch.
Didn't know about that dependency:analyze thing. Looks very interesting !

Reply:
Some jars are wrongly detected as you warned but it has slimmed down a lot:
```
[WARNING] Unused declared dependencies found:
[WARNING]    org.nuxeo.runtime:nuxeo-runtime-mongodb:test-jar:tests:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-userworkspace-types:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-userworkspace-core:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.core:nuxeo-core-cache:test-jar:tests:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-search-core:jar:11.2-SNAPSHOT:test
[WARNING]    org.nuxeo.ecm.platform:nuxeo-platform-tag-core:jar:11.2-SNAPSHOT:test
```
and no more 
```[WARNING] Used undeclared dependencies found:```